Userland/Libraries/LibCore/File.cpp

```diff
@@ -572,9 +572,10 @@ Optional<String> File::resolve_executable_from_environment(StringView filename)
     }
 
     auto const* path_str = getenv("PATH");
-    StringView path { path_str, strlen(path_str) };
-
-    if (path.is_empty())
+    StringView path;
+    if (path_str)
+        path = { path_str, strlen(path_str) };
+    else
```

Comment:
The suggestion wasn't to make this `else`, but to keep it `if (path.is_empty())`, so that we still fall into this branch if `PATH` is the empty string. i.e.:

```c++
    auto const* path_str = getenv("PATH");
    StringView path;
    if (path_str)
        path = { path_str, strlen(path_str) };
    if (path.is_empty())
        path = DEFAULT_PATH_SV;
```

Reply:
if PATH is an empty string, it seems to me the user's intention is that it's blank.  it makes sense to use a default value if the env var doesn't exist at all, but why would we overwrite it if explicitly set to empty string?

Reply:
I'd rather keep the existing behavior - before this patch, if `PATH` is the empty string, then we'd fall back to the default. If we want to change behavior, and others agree with that change, then that can be a separate patch.